provider/source/src/datetime/patterns.rs

```diff
@@ -107,10 +107,10 @@ impl From<&cldr_serde::ca::Dates> for TimeLengths<'_> {
         let length_combinations_v1 = patterns::GenericLengthPatterns::from(&other.datetime_formats);
         let skeletons_v1 = DateSkeletonPatterns::from(&other.datetime_formats.available_formats);
 
-        let pattern_str_full = other.time_formats.full.get_pattern();
-        let pattern_str_long = other.time_formats.long.get_pattern();
-        let pattern_str_medium = other.time_formats.medium.get_pattern();
-        let pattern_str_short = other.time_formats.short.get_pattern();
+        let pattern_str_full = other.time_skeletons.full.get_pattern();
+        let pattern_str_long = other.time_skeletons.long.get_pattern();
+        let pattern_str_medium = other.time_skeletons.medium.get_pattern();
+        let pattern_str_short = other.time_skeletons.short.get_pattern();
```

Comment:
Suggestion (optional, follow-up OK): This is a bit confusing since `TimeLengths` is only used for `preferred_hour_cycle`. I think there's another issue to clean this up. We should at least add a comment that the actual patterns here are not used anywhere any more.